src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.ts

```diff
@@ -221,18 +236,18 @@ export class ChallengeleaderboardComponent implements OnInit, AfterViewInit {
     const SELF = this;
     return (phaseSplit) => {
       if (SELF.router.url.endsWith('leaderboard')) {
-        SELF.router.navigate(['../' + phaseSplit['phase_split']['id']], {relativeTo: this.route});
-      } else if (SELF.router.url.indexOf(phaseSplit['phase_split']['id']) < 0 && SELF.router.url.split('/').length === 5) {
-        SELF.router.navigate(['../' + phaseSplit['phase_split']['id']], {relativeTo: this.route});
-      } else if (SELF.router.url.indexOf(phaseSplit['phase_split']['id']) < 0 && SELF.router.url.split('/').length === 6) {
-        SELF.router.navigate(['../../' + phaseSplit['phase_split']['id']], {relativeTo: this.route});
-      } else {
-        SELF.selectedPhaseSplit = phaseSplit;
-        if (SELF.selectedPhaseSplit['phase_split']) {
-          SELF.fetchLeaderboard(SELF.selectedPhaseSplit['phase_split']['id']);
-        }
+        SELF.router.navigate([phaseSplit['id']], {relativeTo: this.route});
```

Comment:
@Sanji515 can you also refer this file in #171?

Reply:
sure, thanks :+1: